src/ui/wxWidgets/TreeCtrl.cpp

```diff
@@ -99,6 +101,24 @@ BEGIN_EVENT_TABLE( TreeCtrl, wxTreeCtrl )
   EVT_TREE_END_LABEL_EDIT( ID_TREECTRL, TreeCtrl::OnEndLabelEdit )
   EVT_TREE_END_LABEL_EDIT( ID_TREECTRL_1, TreeCtrl::OnEndLabelEdit )
   EVT_TREE_KEY_DOWN( ID_TREECTRL, TreeCtrl::OnKeyDown )
+  /*
+    In Linux environments context menus appear on Right-Down mouse click.
+    Which mouse click type (Right-Down/Right-Up) is the right one on a platform
+    is considered by 'EVT_CONTEXT_MENU(TreeCtrl::OnContextMenu)' which doesn't
+    work for wxTreeCtrl prior wx version 3.1.1. See also the following URL.
+    https://github.com/wxWidgets/wxWidgets/commit/caea08e6b2b9e4843e84c61abc879880a08634b0
+  */
+#if wxCHECK_VERSION(3, 1, 1)
+  EVT_CONTEXT_MENU(TreeCtrl::OnContextMenu)
+#else
+#ifdef __WINDOWS__
```

Comment:
Isn't the whole wxWidgets part Linux-specific?

Reply:
Actually, no.
- The wx code also runs on Mac
- It's also built and run under Windows as an exercise in portability.

Reply:
Would be great if someone could test the implementation with wxWidgets 3.1.1 and on platforms other than Linux.